doc/source/data/working-with-tensors.rst

```diff
@@ -33,40 +33,51 @@ Batches of fixed-shape tensors
 
 If your tensors have a fixed shape, Ray Data represents batches as regular ndarrays.
 
-.. doctest::
+.. testcode::
+
+    import ray
+    ds = ray.data.read_images("s3://anonymous@air-example-data/digits")
+    batch = ds.take_batch(batch_size=32)
+    batch["image"].shape
+    batch["image"].dtype
+    dtype('uint8')
+
+.. testoutput::
 
-    >>> import ray
-    >>> ds = ray.data.read_images("s3://anonymous@air-example-data/digits")
-    >>> batch = ds.take_batch(batch_size=32)
-    >>> batch["image"].shape
     (32, 28, 28)
-    >>> batch["image"].dtype
     dtype('uint8')
 
+
 Batches of variable-shape tensors
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 If your tensors vary in shape, Ray Data represents batches as ragged arrays.
 
-.. doctest::
+.. testcode::
+
+    import ray
+    ds = ray.data.read_images("s3://anonymous@air-example-data/AnimalDetection")
+    batch = ds.take_batch(batch_size=32)
+    batch["image"].shape
+    batch["image"].dtype
+
+.. testoutput::
 
-    >>> import ray
-    >>> ds = ray.data.read_images("s3://anonymous@air-example-data/AnimalDetection")
-    >>> batch = ds.take_batch(batch_size=32)
-    >>> batch["image"].shape
     (32,)
-    >>> batch["image"].dtype
     dtype('O')
 
 Elements of ragged arrays are regular ndarrays.
 
-.. doctest::
+.. testcode::
 
-    >>> batch["image"][0].dtype
+    batch["image"][0].dtype
     dtype('uint8')
-    >>> batch["image"][0].shape  # doctest: +SKIP
+    batch["image"][0].shape  # doctest: +SKIP
+    batch["image"][3].shape  # doctest: +SKIP
+
+.. testoutput::
+
     (375, 500, 3)
-    >>> batch["image"][3].shape  # doctest: +SKIP
     (333, 465, 3)
```

Comment:
This example emphasizes object representations, so I intentionally chose to use doctest-style. I felt it illustrated the API better:

**doctest-style**
<img width="776" alt="image" src="https://github.com/ray-project/ray/assets/26107013/ca7a6d7e-0d3f-49d9-a68c-3a53492ce21f">

**code-block-style**
<img width="774" alt="image" src="https://github.com/ray-project/ray/assets/26107013/67656b6f-76a4-43d7-938e-2c5f8204d660">